src/Core/Framework/DataAbstractionLayer/Dbal/EntityForeignKeyResolver.php

```diff
@@ -241,12 +241,14 @@ private function fetchAssociation(
             return [];
         }
 
-        // create flat list for single primary key entities
+        // create flat list for single primary key entities, if it is 'id'
         if ($primaryKeys->count() === 1) {
             /** @var Field $pk */
             $pk = $primaryKeys->first();
             $property = $pk->getPropertyName();
-            $affected = array_column($affected, $property);
+            if ($property === 'id') {
```

Comment:
this can not be the case because we fetch the field with the property name as alias:

https://github.com/shopware/platform/pull/3098/files#diff-d8cec99d252c25d8e2122aeb578ba052e8d9755969629e5ede0116c76f2a0d4eR216